### DISCUSSION TITLE
TypeError: undefined is not an object (evaluating 'audio.currentTime = 0') - on some files!

### DISCUSSION THREAD

**Discussion by inkbear** [jspsych/jsPsych#1247] [Q&A]
I have been loading mp3 files successfully both on local host and the server until recently when I added a new directory of files. It now seems that any new audio file does not load into the JS though I can access the file in the browser directly. 

The error int he JS console is:
TypeError: undefined is not an object (evaluating 'audio.currentTime = 0')

This error only occurs when loading the new files which I have obtained from several different sources. I am suspicious of MacOS Catalina. More testing to follow...

My JS includes:
```
    <script src="jspsych/jspsych.js"></script>
    <script src="jspsych/plugins/jspsych-html-keyboard-response.js"></script>
    <script src="jspsych/plugins/jspsych-audio-button-response.js"></script>
    <script src="storage/cct/english/v2/trial_info.js"></script>
    <link href="jspsych/css/jspsych.css" rel="stylesheet" type="text/css"></link>
```
My trial_info.js segment changing the directory to a v2 copy fails. Makes me think permissions.

```
var train_info = [
    {
        target: "horse",
        distractor: "cow",
        target_image: "storage/cct/english/v1/horse.jpeg",
        distractor_image: "storage/cct/english/v1/cow.jpeg",
        target_image_button: '<img src="storage/cct/english/v1/horse.jpeg" height="250px"/>',
        distractor_image_button: '<img src="storage/cct/english/v1/cow.jpeg" height="250px"/>',
        prompt_audio: "storage/cct/english/v1/touch_horse.mp3",
        feedback_audio: "storage/cct/english/v1/horse.mp3",
    },
```

**Comment by jodeleeuw** [comment]
Hi @inkbear.

Do you see any information in the network tab of the developer tools that would suggest the files aren't able to load?

**Reply by inkbear** [reply]
THANK YOU! You set me on the right path. I could see NO audio was getting downloaded so I could only conclude the audio I was hearing was cached. Since everything was supposed to be preloaded I looked at the array of objects and how it was getting preloaded. Sure enough, when I expanded the set to train and test I doubled the call to preload_audio and preload_images.

Before:

```
    /* start the experiment */
    jsPsych.init({
        timeline: timeline,
        preload_audio: train_audio,
        preload_audio: test_audio,
        preload_images:train_images,
        preload_images: test_images,
```

After:
```
    // create a single array for all the images and a single array for all the audio
    var all_audio = train_audio.concat(test_audio);
    var all_images = train_images.concat(test_images);

    /* start the experiment */
    jsPsych.init({
        timeline: timeline,
        preload_audio: all_audio,
        preload_images: all_images,
```
Thank you again! Back on building the full experiment.
